### PR TITLE
create a dedicated mozfest 404 page, triggering on mozillafestival.org

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/3275

I'm not actually sure how this can best be tested: the 404 routing now relies on host filtering, which of course ideally we want to do based on the `sites` entries in wagtail. This code does not do that right now (even better would be if wagtail let you specify a 404 in its sites admin, but we're not that far along yet).

The easiest way to test this right now is to check out the branch, chack the `mozillafestival.org` host to `localhost` instead in the new `wagtailpages/views.py` 404 handler, and then loading up a URL you know is bogus **with DEBUG=False** because I haven't figured out a way to make 404 pages show up propertly with DEBUG=True yet - there's a number of supposedly working ways to make that happen by changing the 404 url handling, but I haven't managed to make any of them work.

I suspect this is a WIP PR because it'd be nice to still add in that `sites` filtering, and ideally, 404-with-DEBUG=True, but given that mozfest currently has no 404 of its own at all, it might be worth landing first with immediate followup for those issues.

**A concern**

The package-lock.json is showing _a ton_ of dependencies suddenly resolving on `http` rather than `https`. I have no idea why, npm install manages that for you, but the fact that it's suddenly no longer https is ... worrisome?

**A confusion**

Travis is failing on something that I'm not sure I understand. I can't reproduce it locally (`inv test` passes just fine) and the error messages are... puzzling

### CODE REVIEW

**Comment by Pomax** [comment]
Screen Shot 2019-07-08 at 5 46 04 PM

@gideonthomas @kristinashu, I based the 404 off of what we have on the foundation site - do we need anything more tailored to mozfest or does this look fine?

**Comment by kristinashu** [comment]
Looks good to me (as long as the "Go to home page" button links to the mozfest home page)!

**Comment by Pomax** [comment]
yep, it simply points to `/` on whatever domain is loading it, so on mozillafestival.org that's what it'll point to.

File: network-api/networkapi/wagtailpages/views.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    It would be great if we could pull the "which domain uses
+    which 404 template" information from the wagtail "sites"
+    configuration, but there is no way to know which template
+    belongs to which site, as "a site" is not tied to "a django
+    app" in the wagtail way of things.
+    """
+    if request.site.hostname == 'mozillafestival.org':
+        html = render(request, 'mozfest/404.html')
+        return HttpResponseNotFound(html)
```
Comment (patjouk) [inline network-api/networkapi/wagtailpages/views.py]:
```suggestion
        return HttpResponseNotFound(html.content)
```

cf https://code.djangoproject.com/ticket/29238

File: network-api/networkapi/wagtailpages/views.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    else:
+        html = render(request, '404.html')
+        return HttpResponseNotFound(html)
```
Comment (patjouk) [inline network-api/networkapi/wagtailpages/views.py]:
```suggestion
        return HttpResponseNotFound(html.content)
```

**Comment by patjouk** [comment]
Can you make sure to use postgresql locally? It's what we're using in prod and CI and it would have helped you reproduce this bug locally.

**Comment by Pomax** [comment]
I was, which is why I was so confused. I run 10.6 with a copy of the staging db because I can't really do meaningful dev work otherwise, so I don't understand why this wasn't throwing any errors before. I blasted away my repo, and my db, then recloned and checked out this branch, and did a new staging db copy, and now the error does show up locally so I have _no_ idea why I wasn't see it before.

Thanks for helping debug this.

**Comment by patjouk** [comment]
Those DBs are not connected: [Django creates](https://docs.djangoproject.com/en/2.2/topics/testing/overview/#the-test-database) a new empty db for running tests.

**Pull request merged**